File: leetcode/js/53.Maximum Subarray.js
class Solution{
  
    static maxSubArray(nums){
      let currSum =0;
      let max = Number.MIN_SAFE_INTEGER
  
      for(let i=0;i<nums.length;i++){
         
        currSum = Math.max(currSum + nums[i],nums[i])
        max = Math.max(currSum,max)
      }
      
      return max
  
    }
  
  }
  
console.log(Solution.maxSubArray([1,2,3]))

File: leetcode/c++/53.maximumSubArray.cpp
#include <iostream>
#include <limits.h>
#include <cmath>
using namespace std;
int maxSubArray(int nums[],int length){
int currSum = 0;
int maximum = INT_MIN;
for (int i = 0; i < length; i++) {
currSum = max(currSum + nums[i], nums[i]);
maximum = max(currSum, maximum);
}
return maximum;
}
int main(){
int a[]{4,-1,2,1};
int len = sizeof(a)/sizeof(a[0]);
cout << maxSubArray(a,len)<<endl;
return 0;
}